generated/nifpga/nifpga_service.cpp

```diff
@@ -2852,6 +4828,34 @@ namespace nifpga_grpc {
     }
   }
 
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFpgaService::BeginWriteU8(::grpc::ServerContext* context, const BeginWriteU8Request* request, BeginWriteU8Response* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto session_grpc_session = request->session();
+      NiFpga_Session session = session_repository_->access_session(session_grpc_session.name());
+      uint32_t control = request->control();
+
+      MonikerWriteU8Data* data = new MonikerWriteU8Data();
+            data->session = session;
+      data->control = control;
+      data->library = std::shared_ptr<NiFpgaLibraryInterface>(library_);
+      
+      ni::data_monikers::Moniker* moniker = new ni::data_monikers::Moniker();
+      ni::data_monikers::DataMonikerService::RegisterMonikerInstance("MonikerWriteU8", data, *moniker);
+      response->set_allocated_moniker(moniker);
+      response->set_status(0);
```

Comment:
Is ownership of this moniker pointer being transferred to RegisterMonikerInstance? If so using a std::unique_ptr and then releasing it to the RegisterMonikerInstance would help show that else it again seems like a memory leak.

Reply:
I have transferred the ownership of the moniker to the set_allocated_moniker, which now takes the responsibility and ensure its memory management.